docs/conf.py

```diff
@@ -2,306 +2,172 @@
-    argv = [
-        '--force',
-        '--no-toc',
-        # '--separate',
-        '--module-first',
-        '--output-dir',
-        './api/modules',
-        # this is expected to be run from the `docs` directory.  Its possible
-        # to set it up to run from other places, but so far this has worked for
-        # us
-        '../src/py-opentimelineio/opentimelineio',
-    ] + ignore_paths
+autodoc_default_options = {
+    'undoc-members': True
+}
 
-    from sphinx.ext import apidoc
-    apidoc.main(argv)
+# -- Options for MySt-Parser ---------------------------------------------------------------
+# https://myst-parser.readthedocs.io/en/latest/sphinx/reference.html
+
+myst_heading_anchors = 5
+
+# -- Custom ----------------------------------------------------------------------------
+
+def process_signature(
+    app,
+    what: str,
+    name: str,
+    obj: object,
+    options: dict[str, str],
+    signature: str,
+    return_annotation,
+):
+    """This does several things:
+    * Removes "self" argument from a signature. Pybind11 adds self to
+      method arguments, which is useless in a python reference documentation.
+    * Handles overloaded methods/functions by using the docstrings generated
+      by Pybind11. Pybind11 adds the signature of each overload in the first function's
+      signature. So the idea is to generate a new signature for each one instead.
+    """
+    signatures = []
+    isClass = what == "class"
+
+    # This block won't be necessary once https://github.com/pybind/pybind11/pull/2621
+    # gets merged in Pybind11.
+    if signature or isClass:
+        docstrLines = obj.__doc__ and obj.__doc__.split("\n") or []
+        if not docstrLines or isClass:
+            # A class can have part of its doc in its docstr or in the __init__ docstr.
+            docstrLines += (
+                obj.__init__.__doc__ and obj.__init__.__doc__.split("\n") or []
+            )
+
+        # This could be solidified by using a regex on the reconstructed docstr?
+        if len(docstrLines) > 1 and "Overloaded function." in docstrLines:
+            # Overloaded function detected. Extract each signature and create a new
+            # signature for each of them.
+            for line in docstrLines:
+                nameToMatch = name.split(".")[-1] if not isClass else "__init__"
+
+                # Maybe get use sphinx.util.inspect.signature_from_str ?
+                if match := re.search(f"^\d+\.\s{nameToMatch}(\(.*)", line):
+                    signatures.append(match.group(1))
+        elif signature:
+            signatures.append(signature)
+
+    signature = ""
+
+    # Remove self from signatures.
+    for index, sig in enumerate(signatures):
+        newsig = re.sub("self\: [a-zA-Z0-9._]+(,\s)?", "", sig)
+        signatures[index] = newsig
+
+    signature = "\n".join(signatures)
+    return signature, return_annotation
+
+
+def process_docstring(
```

Comment:
same question ~ I assume this routine as well is just about removing .self.

Reply:
or do these routines serve to remove the need for api_doc?

Reply:
This one removes self from the docstrings in case where a function/method is overloaded. For example, see https://jcmorin-opentimelineio.readthedocs.io/en/latest/api/python/opentimelineio.core.html#opentimelineio.core.flatten_stack. The docstring doesn't contain `self`  because it was removed by `process_docstring`.

So I guess I need to add a little explanation for this one.

Reply:
I added a comment in the function to clarify.